### PR TITLE
Adding request/response protocol for TTL updates

### PR DESCRIPTION
**Pull request by vgkholla**

### CODE REVIEW

File: ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateResponse.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    int correlationId = stream.readInt();
+    String clientId = Utils.readIntString(stream);
+    ServerErrorCode error = ServerErrorCode.values()[stream.readShort()];
+    return new TtlUpdateResponse(correlationId, clientId, error);
+  }
+
+  @Override
+  public String toString() {
+    return "TtlUpdateResponse[" + "ServerErrorCode=" + getError() + "]";
```
Comment (jsjtzyy) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateResponse.java]: Looks like unit test doesn't cover `toString()` method.

File: ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  @Override
+  public long sizeInBytes() {
+    // header + blobId
+    // expires time ms
+    // op time ms
+    return super.sizeInBytes() + blobId.sizeInBytes() + 2 * Long.BYTES;
+  }
+
+  @Override
+  public String toString() {
```
Comment (jsjtzyy) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java]: Looks like unit test doesn't cover `toString()` method.

File: ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+/**
+ * Request to update the TTL of a blob
+ */
+public class TtlUpdateRequest extends RequestOrResponse {
+
+  static final short TTL_UPDATE_REQUEST_VERSION_1 = 1;
+  private final static short CURRENT_VERSION = TTL_UPDATE_REQUEST_VERSION_1;
+
+  private final short version;
```
Comment (pnarayanan) [inline ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java]: `version` does not serve any purpose in this class as this is duplicated in the super class (where it is a protected variable). Serialization ignores this field for the same reason. This is true for `DeleteRequest` as well. Better get rid of this from both classes.

File: ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateResponse.java (hunk 1 above)
Comment (vgkholla) [reply ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateResponse.java]: Yeah - we usually don't have tests for `toString()` (none of the other classes do either)

File: ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java (hunk 2 above)
Comment (vgkholla) [reply ambry-protocol/src/main/java/com.github.ambry.protocol/TtlUpdateRequest.java]: Yeah - we usually don't have tests for `toString()` (none of the other classes do either)

**Pull request merged**